### PR TITLE
Filled doc strings in bmgui package

### PR DESCRIPTION
**Pull request by TickyWill**
Description proposal for classes and functions of 'bmgui' package.

### CODE REVIEW

File: bmgui/gui_utils.py [hunk 1]
```diff
@@ -34,22 +34,35 @@
...
     frame = self.frames[page_name]
     frame.tkraise()
 
 
 def set_page_title(self, master, page_name, institute, datatype = None):
-    """
+    """Sets the page title of the 'page_name' page.
+
+    Args:
+        self (instense): Instense where is located 'page_name' page.
```
Comment (BaptisteRefalo) [inline bmgui/gui_utils.py:57]: Comment applicable elsewhere: it is "instance" and not "instense"

File: bmgui/gui_utils.py [hunk 2]
```diff
@@ -34,22 +34,35 @@
...
+        master (class): `bmgui.main_page.AppMain` class.
+        page_name (str): Name of 'page_name' page.
+        institute (str): Institute name.
+        datatype (str): Data combination type from corpuses databases (default = None). 
+
+    Returns:
+        None.
```
Comment (BaptisteRefalo) [inline bmgui/gui_utils.py:64]: Not necessary

File: bmgui/gui_utils.py [hunk 3]
```diff
@@ -87,7 +100,16 @@ def set_page_title(self, master, page_name, institute, datatype = None):
...
 def set_exit_button(self, master):
-    """ """
+    """Sets exit button on any page of 'master'.
+
+    Args:
+        self (instense): Instense where is located 'master'.
+        master (class): `bmgui.main_page.AppMain` class.
+
+    Returns:
+        None.
```
Comment (BaptisteRefalo) [inline bmgui/gui_utils.py:110]: Not necessary

File: bmgui/parse_corpus_page.py [hunk 4]
```diff
@@ -37,12 +37,18 @@
...
+    Args:
+        year (str): Corpus year defined by 4 digits.
+        wos_r (bool): Status of WoS raw-data file.
+        wos_p (bool): Status of WoS parsing files.
+        scopus_r (bool): Status of Scopus raw-data file.
+        scopus_p (bool): Status of Scopus parsing files.
+        concat (bool) : Status of concatenation and deduplication files.
+
+    Returns:
+        None.
```
Comment (BaptisteRefalo) [inline bmgui/parse_corpus_page.py:51]: A class does not return anything (must be removed)

File: bmgui/parse_corpus_page.py [hunk 5]
```diff
@@ -92,14 +98,12 @@ def efface(self):
 
 def _create_table(self, master, pos_x_init):
     """The internal function `_create_table` creates the column names
-    of the table used by 'parse_corpus_page' to display which files
-    of the parsing treatment are available in the working folder.
-    The positions of the table items in the gui window 'root' are set
-    using the argument 'pos_x_init', and the tk window general properties
-    as 'master' class variables.
+    of the table displaing which files of the parsing step are available
```
Comment (BaptisteRefalo) [inline bmgui/parse_corpus_page.py:101]:
```suggestion
    of the table displaying which files of the parsing step are available 
```

File: bmgui/update_if_page.py [hunk 6]
```diff
@@ -155,27 +186,24 @@ def _launch_update_pub_if(institute,
...
+    Args:
+        self (instense): Instense where consolidation page will be created.
+        master (class): `bmgui.main_page.AppMain` class.
+        page_name (str): Name of consolidation page.
+        institute (str): Institute name.
+        bibliometer_path (path): Full path to working folder.
+        datatype (str): Data combination type from corpuses databases. 
+
+    Returns:
+        None.
```
Comment (BaptisteRefalo) [inline bmgui/update_if_page.py:206]: Not necessary

**Review by BaptisteRefalo — COMMENTED** [review]
I try to be as thorough as possible, but I may have missed some typos :)
If a function has no "return" clause, the "return" inside the docstring is not needed

File: bmfuncts/consolidate_pub_list.py [hunk 7]
```diff
@@ -169,13 +177,29 @@ def _get_dpt_key(dpt_raw):
...
-    """ Create and store an Excel file under 'excel_dpt_path' for the department labelled 'dpt'.
-    The OPTs of the choosen department are added in a new column named 'otp_alias'.
-    A list data validation rules is added to each celles of the column
-    'otp_alias'. The data frame column are renamed using 'otp_col_list'. The Excel frame is
-    configurated by the `mise_en_page` function.
+    """Creates the file for setting OTP attribute of publications by the user 
+    for the Institute department labelled 'dpt'.     
+    A new column named 'otp_alias' is added to the dataframe 'dpt_df'.
+    A list data validation rules is added to each cells of the column
+    'otp_alias' based on the list of OPTs of the department given
```
[the remark this answers is not in the document]
Comment (TickyWill) [reply bmfuncts/consolidate_pub_list.py:186]: Yes !! Thanks

[the remark this answers is not in the document]
Comment (TickyWill) [reply bmfuncts/consolidate_pub_list.py:186]: Correction done. Thanks

File: bmfuncts/consolidate_pub_list.py [hunk 8]
```diff
@@ -457,9 +559,20 @@ def _format_missing_df(results_df, common_args_tup, add_cols):
 
 
 def get_if_db(institute, org_tup, bibliometer_path):
-    """
+    """The function `get_if_db` builds a dataframe of impact-factors
+    of he Institute.
```
[the remark this answers is not in the document]
Comment (TickyWill) [reply bmfuncts/consolidate_pub_list.py:563]: Correction done. Thanks

File: bmfuncts/merge_pub_employees.py [hunk 9]
```diff
@@ -527,42 +374,296 @@ def _build_filt_authors_inst(inst_col_list, main_inst_idx, main_status):
...
-    The job type is get from the employee information available
-    in 3 particular columns which names
-    are defined by the global 'EMPLOYEES_USEFUL_COLS'
-    at keys 'category', 'status' and 'qualification'.
-    The name of the new column is defined by
-    the global 'COL_NAMES_BONUS' at key 'author_type'.
+    """The function `_add_author_job_type` adds a new column 
+    containing the job type for each author of the publications 
+    list with one row per author and saves it as Excel file.
+    The job type is get from the employee information available
```
[the remark this answers is not in the document]
Comment (TickyWill) [reply bmfuncts/merge_pub_employees.py:653]: Correction done. Thanks

File: bmfuncts/merge_pub_employees.py [hunk 10]
```diff
@@ -972,21 +1074,38 @@ def _change_col_names(institute, org_tup, submit_path, orphan_path):
...
+
+    Args:
+        institute (str): Institute name.
+        org_tup (tup): Contains Institute parameters.
+        bibliometer_path (path): Full path to working folder.
+        corpus_year (str): Contains the corpus year defined by 4 digits.
+        verbose (bool): Status of prints (default = False).
+
+    Returns:
+        None.
```
[the remark this answers is not in the document]
Comment (TickyWill) [reply bmfuncts/merge_pub_employees.py:1106]: Taken into account. Thanks

File: bmgui/gui_utils.py (hunk 2 above)
Comment (TickyWill) [reply bmgui/gui_utils.py:64]: Taken into account. Thanks

File: bmgui/gui_utils.py (hunk 3 above)
Comment (TickyWill) [reply bmgui/gui_utils.py:110]: Taken into account. Thanks

File: bmgui/parse_corpus_page.py (hunk 4 above)
Comment (TickyWill) [reply bmgui/parse_corpus_page.py:51]: Taken into account. Thanks

File: bmgui/parse_corpus_page.py (hunk 5 above)
Comment (TickyWill) [reply bmgui/parse_corpus_page.py:101]: Correction done. Thanks

**Pull request merged**